cmd/kubeapps-apis/plugins/resources/v1alpha1/server.go

```diff
@@ -216,8 +220,17 @@ func (s *Server) GetResources(r *v1alpha1.GetResourcesRequest, stream v1alpha1.R
 			continue
 		}
 
-		watcher, err := dynamicClient.Resource(gvr).Namespace(namespace).Watch(stream.Context(), metav1.ListOptions{})
+		listOptions := metav1.ListOptions{
+			FieldSelector: fmt.Sprintf("metadata.name=%s", ref.GetName()),
```

Comment:
This is the only way I can see to watch a particular resource using the client-go's Watch.